runtime/src/accounts_index.rs

```diff
@@ -1136,22 +1136,28 @@ impl<T: IndexValue> AccountsIndex<T> {
         read_lock.slot_list_mut(pubkey, user)
     }
 
+    /// Remove keys from the account index if the key's slot list is empty.
+    /// Returns the keys that were removed from the index. These keys should not be accessed again in the current code path.
+    #[must_use]
     pub fn handle_dead_keys(
         &self,
         dead_keys: &[&Pubkey],
         account_indexes: &AccountSecondaryIndexes,
-    ) {
+    ) -> HashSet<Pubkey> {
+        let mut pubkeys_removed_from_accounts_index = HashSet::default();
```

Comment:
I considered it. I figured it was useful to know it was a HashSet since this function works on it. It is clearer for the return values of the functions what the INTENTION of the hashset is to use the type. The name of the variable here indicates the intention. I liked using the type here to make it clear what it actually is here.